3.7/ubuntu/Dockerfile

```diff
@@ -0,0 +1,271 @@
+# Record OpenSSL artefacts size
+RUN du -kh -d 1 $OPENSSL_INSTALL_DIR $OPENSSL_CONFIG_DIR >> /OPENSSL_BUILD_ARTEFACTS
+
+# Use the latest stable Erlang/OTP release
+ARG OTP_VERSION="21.2.2"
+ARG OTP_SOURCE_URL="https://github.com/erlang/otp/archive/OTP-$OTP_VERSION.tar.gz"
+ARG OTP_PATH=/usr/local/src/otp-$OTP_VERSION
+ARG OTP_INSTALL_DIR=/opt/otp
+RUN mkdir -p $OTP_PATH $OTP_INSTALL_DIR
+# TODO add PGP checking when the feature will be added to Erlang/OTP's build system
+# http://erlang.org/pipermail/erlang-questions/2019-January/097067.html
+ARG OTP_SOURCE_SHA256="41b1c3a8343218157a683776e71d80a05ac4eb4019a90a760846608d78817690"
+
+# Download, verify & extract OTP_SOURCE
+# Fail fast if SOURCE URL returns HTTP errors
+RUN curl --verbose --head --fail --fail-early $OTP_SOURCE_URL 1>/dev/null && \
+    curl --verbose --location --silent --fail --fail-early --output $OTP_PATH.tar.gz $OTP_SOURCE_URL && \
+    echo "$OTP_SOURCE_SHA256  $OTP_PATH.tar.gz" > $OTP_PATH.sha256 && \
+    sha256sum --check --strict $OTP_PATH.sha256 && \
+    tar -xvf $OTP_PATH.tar.gz --directory $OTP_PATH --strip-components=1
+
+# Configure Erlang/OTP for compilation, disable unused features & applications
+# http://erlang.org/doc/applications.html
+# ERL_TOP is required for Erlang/OTP makefiles to find the absolute path for the installation
+RUN cd $OTP_PATH && \
+    export ERL_TOP=$OTP_PATH && \
+    ./otp_build autoconf && \
+    export CFLAGS="$(dpkg-buildflags --get CFLAGS)" && \
+    ./configure \
+      --host="$(dpkg-architecture --query DEB_HOST_GNU_TYPE)" \
+      --build="$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)" \
+      --prefix=$OTP_INSTALL_DIR \
+      --disable-sctp \
+      --disable-silent-rules \
+      --disable-dynamic-ssl-lib \
+      --with-ssl=$OPENSSL_INSTALL_DIR \
+      --enable-clock-gettime \
+      --enable-hybrid-heap \
+      --enable-kernel-poll \
+      --enable-shared-zlib \
+      --enable-threads \
+      --enable-smp-support \
+      --enable-hipe \
+      --with-microstate-accounting=extra \
+      --without-odbc \
+      --without-diameter \
+      --without-erl_interface \
+      --without-ftp \
+      --without-jinterface \
+      --without-megaco \
+      --without-ssh \
+      --without-tftp \
+      --without-wx \
+      --without-debugger \
+      --without-dialyzer \
+      --without-et \
+      --without-observer \
+      --without-reltool \
+      --without-common_test \
+      --without-eunit \
+      --without-edoc \
+      --without-erl_docgen
+
+# Compile & install Erlang/OTP
+RUN cd $OTP_PATH && export ERL_TOP=$OTP_PATH && make -j "$(getconf _NPROCESSORS_ONLN)" GEN_OPT_FLGS="-O2 -fno-strict-aliasing" && make install
+
+# Delete all src dirs to keep the final image size down, they add an extra 19MB
+# TODO Could this be done better?
+RUN find $OTP_INSTALL_DIR/lib/erlang/lib -maxdepth 2 -mindepth 2 -name "src" -exec rm -fr {} \;
```

Comment:
Doesn't this amount to essentially `rm -rf $OTP_INSTALL_DIR/lib/erlang/lib/*/src`?

Reply:
Good catch, I've left my exploration shovel in there 😄